resources/assets/js/components/H2Components/Dialog.tsx

```diff
@@ -0,0 +1,220 @@
+import * as React from "react";
+import {
+  h2ComponentDialogLoad,
+  h2ComponentDialogLoadResize,
+  h2ComponentDialogEnableTrigger,
+} from "@hydrogen-design-system/system/dist/import/latest/components/dialog/scripts/dialog";
+import { defineMessages, useIntl } from "react-intl";
+import { GeneralProps, GeneralBtnProps } from "./utils";
+
+const messages = defineMessages({
+  closeDialog: {
+    id: "hydrogen.dialog.close",
+    defaultMessage: "Close",
+  },
+});
+
+interface DialogContext {
+  /** The dialogs id. */
+  id?: string;
+}
+
+const DialogContext = React.createContext<DialogContext | undefined>(undefined);
+
+/**
+ * This Context hook allows our child components to easily reach
+ * into the Tabs context and get the pieces it needs.
+ *
+ * Bonus: it even makes sure the component is used within a
+ * Dialog component!
+ */
+const useDialogContext = (): Partial<DialogContext> => {
+  const context = React.useContext(DialogContext);
+  if (!context) {
+    throw new Error("This component must be used within a <Dialog> component.");
+  }
+  return context;
+};
+
+const Actions: React.FunctionComponent<GeneralProps> = (props) => {
+  useDialogContext(); // Ensures sub-component can only be used within the Dialog component.
+  const { children } = props;
+  return (
+    <div data-h2-dialog-actions {...props}>
+      {children}
+    </div>
+  );
+};
+
+const ActionBtn: React.FunctionComponent<GeneralBtnProps> = (props) => {
+  const { id = "dialog" } = useDialogContext();
```

Comment:
Is this setting "dialog" as a default value? It might be simpler to make id non-optional in DialogContext, and if there needs to be a default value, it can be defined just once in the `Dialog` component where it's passed into the context provider.